libnmstate/netapplier.py

```diff
@@ -312,12 +316,20 @@ def assert_ifaces_state(ifaces_desired_state, ifaces_current_state):
             ifaces_desired_state[ifname], iface_cstate)
 
         _cleanup_iface_ethernet_state_sanitize(iface_dstate, iface_cstate)
+        _sort_ip_addresses(iface_dstate)
 
         if iface_dstate != iface_cstate:
             raise DesiredStateIsNotCurrentError(
                 'desired: {}\ncurrent: {}'.format(iface_dstate, iface_cstate))
 
 
+def _sort_ip_addresses(iface):
+    for family in ('ipv4', 'ipv6'):
+        if iface.get(family, {}).get('address', []):
+            iface[family]['address'] = sorted(iface[family]['address'],
+                                              key=itemgetter('ip'))
```

Comment:
For programmers that do not know functional programming itemgetter is easier to understand - also it has a descriptive name IMHO.

Reply:
@EdDev I prefer the lamdba way also, but pylint suggest I should use itemgetter instead.

Reply:
@tyll , I think someone did a few steps too much to give such a simple lambda a name, I rarely seen it in use and surprises me more than helps.
@cathay4t , interesting... what is its reasoning?

Maybe I need to learn like in Java all possible options of lambda structures (?!)... if the basics are strong (lambda), we do not need naming for every little thing... it is in front of us to read.

Reply:
Funny thing is, that if you wanted some small deviation like a default value in case it is missing, then you would have to go back to the basics (lambda).